firedrake/slate/slate.py

```diff
@@ -332,15 +442,12 @@ def __init__(self, form):
         if r not in (0, 1, 2):
             raise NotImplementedError("No support for tensors of rank %d." % r)
 
-        # Checks for positive restrictions on integrals
-        integrals = form.integrals()
-        mapper = CheckRestrictions()
-        for it in integrals:
-            map_integrand_dags(mapper, it)
+        # Remove any negative restrictions and replace with zero
+        _form = map_integrand_dags(RemoveNegativeRestrictions(), form)
```

Comment:
You don't need `form` after this call, so just reassign `form = map_integrand_dags(..., form)`